opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -112,6 +119,82 @@ protected void onSpanFinished(MockSpan mockSpan) {
             }
         };
 
+        Propagator BINARY = new Propagator() {
+            static final int BUFFER_SIZE = 128;
+
+            @Override
+            public <C> void inject(MockSpan.MockContext ctx, Format<C> format, C carrier) {
+                if (carrier instanceof Binary) {
+                    Binary binary = (Binary) carrier;
+                    ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                    ObjectOutputStream objStream = null;
+                    try {
+                        objStream = new ObjectOutputStream(stream);
+                        objStream.writeLong(ctx.spanId());
+                        objStream.writeLong(ctx.traceId());
+
+                        for (Map.Entry<String, String> entry : ctx.baggageItems()) {
+                            objStream.writeUTF(entry.getKey());
+                            objStream.writeUTF(entry.getValue());
+                        }
+                        objStream.flush(); // *need* to flush ObjectOutputStream.
+
+                        byte[] buff = stream.toByteArray();
+                        binary.setInjectBufferLength(buff.length);
+                        binary.injectBuffer().put(buff);
+
+                    } catch (IOException e) {
+                        throw new RuntimeException("Corrupted state");
+                    } finally {
+                        if (objStream != null) {
+                            try { objStream.close(); } catch (Exception e2) {}
+                        }
+                    }
+                } else {
+                    throw new IllegalArgumentException("Unknown carrier");
```

Comment:
I would move this to the top to reduce nesting

Reply:
Will do as well.